### PR TITLE
feat:Modifying the return value of multi permission requests

### PR DESCRIPTION
**Pull request by HDJKER**
feat:多请求返回值类型修改

### CODE REVIEW

**Comment by HDJKER** [comment]
feat:func "checkMultiple" reutrn {Permission[1]:GrantStatus...}

**Comment by HDJKER** [comment]
feat:func "requestMultiple" reutrn {Permission[1]:GrantStatus...}

File: harmony/permissions/src/main/ets/PermissionsModule.ts [hunk 1]
```diff
@@ -42,7 +42,8 @@ export class PermissionsModule extends TurboModule {
    * */
   async checkMultiple(permissions: Array<Permissions>): Promise<Object> {
     let atManager: abilityAccessCtrl.AtManager = abilityAccessCtrl.createAtManager();
-    let grantStatus: GrantStatus[] = [];
+    type PermissionStatusRecord = Record<string, GrantStatus>;
+    let grantStatus:PermissionStatusRecord= {};
```
Comment (Louis-C7) [inline harmony/permissions/src/main/ets/PermissionsModule.ts]: 注意格式，少空格，可合并

File: harmony/permissions/src/main/ets/PermissionsModule.ts [hunk 2]
```diff
@@ -57,8 +58,8 @@ export class PermissionsModule extends TurboModule {
     for (let index = 0; index < permissions.length; index++) {
       // 校验应用是否被授予权限
       try {
-        let Status = await atManager.checkAccessToken(tokenId, permissions[index]);
-        grantStatus.push(this.checkStatus(Status));
+        const Status = await atManager.checkAccessToken(tokenId, permissions[index]);
+        grantStatus[permissions[index]]=this.checkStatus(Status);
```
Comment (Louis-C7) [inline harmony/permissions/src/main/ets/PermissionsModule.ts]: 少空格

File: harmony/permissions/src/main/ets/PermissionsModule.ts [hunk 3]
```diff
@@ -97,20 +98,22 @@ export class PermissionsModule extends TurboModule {
    */
   async requestMultiple(permissions: Array<Permissions>): Promise<Object> {
     let atManager: abilityAccessCtrl.AtManager = abilityAccessCtrl.createAtManager();
-    let resultsStatus: GrantStatus[] = [];
+    type PermissionStatusRecord = Record<string, GrantStatus>;
+    let resultsStatus:PermissionStatusRecord= { };
```
Comment (Louis-C7) [inline harmony/permissions/src/main/ets/PermissionsModule.ts]: 少空格，可合并
let resultsStatus: Record<string, GrantStatus> = {};

File: harmony/permissions/src/main/ets/PermissionsModule.ts [hunk 4]
```diff
@@ -97,20 +98,22 @@ export class PermissionsModule extends TurboModule {
...
     try {
       let context = this.ctx.uiAbilityContext;
       const resultData = await atManager.requestPermissionsFromUser(context, permissions)
       for (let index = 0; index < resultData.authResults.length; index++) {
-        resultsStatus.push(this.checkStatus(resultData.authResults[index]));
+        const status = this.checkStatus(resultData.authResults[index]);
+        resultsStatus[permissions[index]]=status;
```
Comment (Louis-C7) [inline harmony/permissions/src/main/ets/PermissionsModule.ts]: 少空格

File: harmony/permissions/src/main/ets/PermissionsModule.ts [hunk 5]
```diff
@@ -97,20 +98,22 @@ export class PermissionsModule extends TurboModule {
...
       }
       return resultsStatus;
     } catch (err) {
       console.error(`catch err->${JSON.stringify(err)}`);
       return `catch err->${JSON.stringify(err)}`;
     }
   }
-
+
```
Comment (Louis-C7) [inline harmony/permissions/src/main/ets/PermissionsModule.ts]: 无意义修改

File: harmony/permissions/src/main/ets/PermissionsModule.ts [hunk 6]
```diff
@@ -97,20 +98,22 @@ export class PermissionsModule extends TurboModule {
    */
   async requestMultiple(permissions: Array<Permissions>): Promise<Object> {
```
Comment (Louis-C7) [inline harmony/permissions/src/main/ets/PermissionsModule.ts:97]: async requestMultiple(permissions: Array<Permissions>): Promise<Record<string, GrantStatus>？

**Review by Louis-C7 — CHANGES_REQUESTED** [review]

**Review by Louis-C7 — APPROVED** [review]

**Pull request merged**